rest/config.go

```diff
@@ -303,27 +304,42 @@ type invalidDatabaseConfigs struct {
 // addInvalidDatabase adds a db to invalid dbconfig map if it doesn't exist in there yet and will log for it at warning level
 // if the db already exists there we will calculate if we need to log again according to the config update interval
 func (d *invalidDatabaseConfigs) addInvalidDatabase(ctx context.Context, dbname string, cnf DatabaseConfig, bucket string) {
-	configInfo := invalidConfigInfo{
-		configBucketName:    *cnf.Bucket,
-		persistedBucketName: bucket,
-	}
 	d.m.Lock()
 	defer d.m.Unlock()
 	if d.dbNames[dbname] == nil {
 		// db hasn't been tracked as invalid config yet so add it
-		d.dbNames[dbname] = &configInfo
+		d.dbNames[dbname] = &invalidConfigInfo{
+			configBucketName:    *cnf.Bucket,
+			persistedBucketName: bucket,
+			collectionConflicts: cnf.Version == invalidDatabaseConflictingCollectionsVersion,
+		}
+	}
+
+	logMessage := "Must repair invalid database config for %q for it to be usable!"
+	logArgs := []interface{}{base.MD(dbname)}
+
+	// build log message
+	if isBucketMismatch := *cnf.Bucket != bucket; isBucketMismatch {
+		base.SyncGatewayStats.GlobalStats.ConfigStat.DatabaseBucketMismatches.Add(1)
+		logMessage += " Mismatched buckets (config bucket: %q, actual bucket: %q)"
+		logArgs = append(logArgs, base.MD(d.dbNames[dbname].configBucketName), base.MD(d.dbNames[dbname].persistedBucketName))
+	} else if cnf.Version == invalidDatabaseConflictingCollectionsVersion {
+		base.SyncGatewayStats.GlobalStats.ConfigStat.DatabaseCollectionCollisions.Add(1)
+		logMessage += " Conflicting collections detected"
```

Comment:
It's not feasible to do that here, but the warning log that occurred when the rollback set this invalid flag logs what collection(s) caused it.

```
2024-02-29T15:39:54.284Z [WRN] db:c1_db1 db <ud>c1_db1</ud> config rollback would cause collection conflicts (<ud>map[sg_test_0.sg_test_1:c1_db2]</ud>) - marking database as invalid to allow for manual repair -- rest.(*GatewayRegistry).rollbackDatabaseConfig() at config_registry.go:234
2024-02-29T15:39:54.285Z [WRN] Must repair invalid database config for "c1_db1" for it to be usable! Conflicting collections detected -- rest.(*invalidDatabaseConfigs).addInvalidDatabase() at config.go:334
```

Given the rollback that just happened, the customer probably doesn't want to change the non-rolled back database, so I'm not sure there's value in knowing which database this one conflicts with. The rollback is the problem that needs correcitng and we log the collection that needs removing.